### PR TITLE
feat: pod manifest edit feature and option to connect session with node debug pod.

### LINKED ISSUE #3597: Feature: Launch node debug pod and manifest edit functionality in cluster terminal pods.
### 🔖 Feature description

1)Ability to launch node debug pods from cluster terminal.
2)Ability to edit terminal pod and launch terminal pod with the edited manifest.

### 🎤 Pitch

1)Ability to launch node debug pods from cluster terminal.
2)Ability to edit terminal pod and launch terminal pod with the edited manifest.

### ✌️ Solution

image

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#4138

### PR DESCRIPTION
**Pull request by gireesh-devtron**
# Description
Added pod manifest edit feature and option to connect session with node debug pod.

Fixes #3597 

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: client/k8s/application/Application.go [hunk 1]
```diff
@@ -158,20 +159,25 @@ func (impl K8sClientServiceImpl) DeleteResource(ctx context.Context, restConfig
 	}
 	resourceIdentifier := request.ResourceIdentifier
 	var obj *unstructured.Unstructured
+	deleteOptions := metav1.DeleteOptions{}
+	if request.ForceDelete {
+		var gracePeriodSeconds int64 = 0
+		deleteOptions.GracePeriodSeconds = &gracePeriodSeconds
```
Comment (kripanshdevtron) [inline client/k8s/application/Application.go:165]: we can use ``pointer.Int64Ptr`` here

File: pkg/terminal/terminalSesion.go [hunk 2]
```diff
@@ -518,3 +504,25 @@ func getErrorMsg(err string) error {
...
 	return errors1.New(CommandExecutionFailed)
 }
+
+func (impl *TerminalSessionHandlerImpl) RunCmdInRemotePod(req *TerminalSessionRequest, cmds []string) (*bytes.Buffer, *bytes.Buffer, error) {
+	config, client, err := impl.getClientConfig(req)
+	if err != nil {
+		impl.logger.Errorw("error in fetching config", "err", err)
+		return nil, nil, err
+	}
+	impl.logger.Infow("reached getExecutor method call")
```
Comment (kripanshdevtron) [inline pkg/terminal/terminalSesion.go:514]: do we need these, If Yes, then Please change Loglevel to ``Debug``

Comment (gireesh-devtron) [reply pkg/terminal/terminalSesion.go]: done

File: client/k8s/application/Application.go (hunk 1 above)
Comment (gireesh-devtron) [reply client/k8s/application/Application.go]: done

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**